### PR TITLE
Repair IMC functionality

### PR DESCRIPTION
**Pull request by RobinGeens**

### CODE REVIEW

**Review by RobinGeens — COMMENTED** [review]
What is the reasoning behind making the private functions public again?

File: zigzag/cost_model/cost_model.py [hunk 1]
```diff
@@ -689,7 +689,7 @@ def calc_double_buffer_flag(self) -> None:
 
         self.double_buffer_true = double_buffer_true
 
-    def __calc_allowed_and_real_data_transfer_cycle_per_DTL(self):
```
Comment (RobinGeens) [inline zigzag/cost_model/cost_model.py:692]: What is the reasoning behind making the private functions public again?

Comment (JiacongSun) [reply zigzag/cost_model/cost_model.py:692]: These functions need to be called within cost_model_imc when inheriting this class. So, it's better keep them to be public.

https://github.com/KULeuven-MICAS/zigzag/blob/41c237f6e9caef1252cd40900b34298cf9ee55d2/zigzag/cost_model/cost_model_imc.py#L100

Comment (RobinGeens) [reply zigzag/cost_model/cost_model.py:692]: Oh right, I forgot `CostModelEvaluationIMC` inherits from `CostModelEvaluation`. Sorry!

**Comment by JiacongSun** [comment]
Do we have a better method to get the layer operand name of weight?
https://github.com/KULeuven-MICAS/zigzag/blob/460fdefd206d3ab4439cd78875a53a6fca955cd1/zigzag/hardware/architecture/imc_unit.py#L150

**Comment by RobinGeens** [comment]
> Do we have a better method to get the layer operand name of weight?
> 
> https://github.com/KULeuven-MICAS/zigzag/blob/460fdefd206d3ab4439cd78875a53a6fca955cd1/zigzag/hardware/architecture/imc_unit.py#L150

Assuming there is only one constant operand, we usually do `next(layer.constant_operands)`. I find this approach a bit clearer than taking the last element. Note that this will give a `LayerOperand`

**Review by asyms — APPROVED** [review]

**Pull request merged**